src/MICore/Checksum.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MICore
+{
+    public class Checksum
+    {
+#if CORECLR
```

Comment:
the only issue is that mono doesn't have HashAlgorithmName yet. I will comment explaining this.